python/inc/bond/python/converters.h

```diff
@@ -181,12 +181,16 @@ extend_map(Map& map, const boost::python::object& obj)
     typedef typename Map::value_type value_type;
 
     BOOST_ASSERT(PyDict_Check(obj.ptr()));
-    
+
     dict dict(obj);
-        
+
     BOOST_FOREACH(object k,
         std::make_pair(
+#if PY_VERSION_HEX >= 0x03000000
```

Comment:
Create a reference to the one you want, not a copy.
#if PY_VERSION_HEX >= 0x03000000
const auto & keys = dict.keys();
#else
const auto & keys = dict.iterkeys();
#endif